game/scripts/vscripts/heroes/antimage/antimage_spell_shield.lua

```diff
@@ -0,0 +1,179 @@
+-- Author: Shush
+-- Date: 25/11/2017
+
+-------------------------------
+--  Anti Mage's Spell Shield --
+-------------------------------
+lua_antimage_spell_shield = class({})
+LinkLuaModifier("modifier_lua_antimage_spell_shield", "antimage/antimage_spell_shield", LUA_MODIFIER_MOTION_NONE)
+
+function lua_antimage_spell_shield:GetIntrinsicModifierName()
+    return "modifier_lua_antimage_spell_shield"
+end
+
+function lua_antimage_spell_shield:GetAbilityTextureName()
+   return "antimage_spell_shield"
+end
+
+function lua_antimage_spell_shield:GetCooldown(level)
+    if self:GetCaster():HasScepter() then
+        return self:GetSpecialValueFor("scepter_cooldown")
+    end
+
+    return 0
+end
+
+
+-- Spell shield modifier
+modifier_lua_antimage_spell_shield = class({})
+
+function modifier_lua_antimage_spell_shield:IsHidden() return true end
+function modifier_lua_antimage_spell_shield:IsDebuff() return false end
+
+function modifier_lua_antimage_spell_shield:OnCreated()
+    -- Ability properties
+    self.caster = self:GetCaster()
+    self.ability = self:GetAbility()
+
+    -- Ability specials
+    self.spell_shield_resistance = self.ability:GetSpecialValueFor("spell_shield_resistance")    
+
+    -- Initialize table of old spells
+    self.caster.tOldSpells = {}
+    
+    if IsServer() then
+        -- Think, in order to check for scepter. Once a scepter is found, the modifier's interval starts
+        Timers:CreateTimer(function()            
+            if self.caster:HasScepter() then
+                self:StartIntervalThink(0.2)
+            else
+                return 20
+            end
+        end)
+    end
+end
+
+function modifier_lua_antimage_spell_shield:OnRefresh()
+    self:OnCreated()
```

Comment:
I wouldn't run oncreated again, that is for **creation**, just update whatever you need in here.